### DISCUSSION TITLE
github.token can access org docker images but not org nuget packages

### DISCUSSION THREAD

**Discussion by davidshen84** [community/community#36600] [Enterprise]
Hi,

I have a workflow in my organisation's repo. This workflow only exist on a branch I am working on right now. At the top of the workflow, I gave the jobs in the workflow with the following permissions.

```yaml
permissions:
  packages: write
  contents: write
  actions: read
  pull-requests: write
```

In the job steps, I used `github.actor` and `github.token` to login for:

- ghcr.io for docker images
- https://nuget.pkg.github.com/my-enterprise/index.json for nuget packages

In the Dockerfile, the base image is from my organisation. After I made a push and triggers the workflow, the `docker/login-action` was successful, and it is able to start buind the image, which I think it indicates I can pull docker images from my organisation.

In the Dockerfile, I also logged in the organisation nuget repository like below and the reponse was successful.

```
RUN --mount=type=secret,id=actor --mount=type=secret,id=token \
    dotnet nuget update source github \
    --username "$(cat /run/secrets/actor)" --password "$(cat /run/secrets/token)" \
    --store-password-in-clear-text
```

But when the `dotnet build` started, it is not able to find any packages.

If I replace `github.token` with `secrets.MY_PACKAGE_READ_TOKEN` which is a token I created using my account, then everything works.

According to https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry, a `github.token` with `packages: read` should be sufficient to allow me to access my organsation's nuget repository.

Thanks

**Comment by ossanna16** [comment]
Hey there @davidshen84 and thank you for asking a good question! I'm glad you found our community 🙂
